packages/core/lib/debug/printer.js

```diff
@@ -209,10 +209,33 @@ class DebugPrinter {
       this.config.logger.log(DebugUtils.formatStack(step.stack));
       this.config.logger.log("");
     }
-    this.config.logger.log(
-      DebugUtils.formatInstruction(traceIndex + 1, totalSteps, instruction)
-    );
-    this.config.logger.log(DebugUtils.formatPC(step.pc));
+
+    // printout instructions
+    const previousInstructions = 3;
+    const upcomingInstructions = 3;
+    const currentInstructionIndex = instructions.findIndex(instruction => instruction.pc === step.pc);
+    // printout 3 previous instructions
+    if (currentInstructionIndex-previousInstructions+1 > 0) {
```

Comment:
Company style nitpick: Style here is to put spaces around operators. This doesn't really matter, obviuosly; it's purely a consistency thing.  But, that is the style here.

Sorry, I realize this is another thing that you shouldn't really have to worry about. :-/  We used to have `prettier` to handle this for us...  hopefully that's fixed soon.  You'll have to ask @cds-amal what the status of that is.

Btw, I know I said we try to avoid abbreviating variable names here, but if you're doing this a bunch of arithmetic with them, it might help to find *some* suitable way of abbreviating them.  (Like, `currentInstructionIndex` could maybe just be `currentIndex`?  Or something like that?)  Idk, it's a balance, I guess.  Sorry, I realize that's not exactly clear unambiguous instructions, but yeah.  Generally you should err on the side of longer names, but for cases like this it might make sense to pull back on that a little.

Reply:
add spaces around operators, and rename currentInstructionIndex to currentIndex, thx